src/System.Web.HttpUtility/src/System/Web/HttpUtility.cs

```diff
@@ -142,52 +143,70 @@ public static NameValueCollection ParseQueryString(string query, Encoding encodi
             return result;
         }
 
-        public static string HtmlDecode(string s) => HttpEncoder.HtmlDecode(s);
+        [return: NotNullIfNotNull("s")]
+        public static string? HtmlDecode(string? s) => HttpEncoder.HtmlDecode(s);
 
-        public static void HtmlDecode(string s, TextWriter output) => HttpEncoder.HtmlDecode(s, output);
+        public static void HtmlDecode(string? s, TextWriter output) => HttpEncoder.HtmlDecode(s, output);
 
-        public static string HtmlEncode(string s) => HttpEncoder.HtmlEncode(s);
+        [return: NotNullIfNotNull("s")]
+        public static string? HtmlEncode(string? s) => HttpEncoder.HtmlEncode(s);
 
-        public static string HtmlEncode(object value) =>
-            value == null ? null : HtmlEncode(Convert.ToString(value, CultureInfo.CurrentCulture));
+        [return: NotNullIfNotNull("value")]
+        public static string? HtmlEncode(object? value) =>
+            value == null ? null : HtmlEncode(Convert.ToString(value, CultureInfo.CurrentCulture) ?? string.Empty);
```

Comment:
I added the `?? string.Empty` here because I thought it important for usability to be able to have the `NotNullIfNotNull` attribution, and it's only necessary for the case where value is not null but its ToString returns null.  As this isn't a high-performance API, I think it's better to add the fallback.